src/tests/baseservices/RuntimeConfiguration/TestConfig.cs

```diff
@@ -0,0 +1,159 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Reflection;
+using System.Runtime;
+
+using Xunit;
+
+class TestConfig
+{
+    const int Success = 100;
+    const int Fail = 101;
+
+    [Fact]
+    [EnvVar("DOTNET_gcServer", "1")]
+    static int Verify_ServerGC_Env_Enable(string[] _)
+    {
+        return GCSettings.IsServerGC
+            ? Success
+            : Fail;
+    }
+
+    [Fact]
+    [ConfigProperty("DOTNET_gcServer", "0")]
+    static int Verify_ServerGC_Env_Disable(string[] _)
```

Comment:
@AaronRobinsonMSFT, FYI, this test failed on Windows arm64 checked https://helix.dot.net/api/2019-06-17/jobs/1a1caea9-5980-4759-a7d1-da30e1a225a6/workitems/baseservices.RuntimeConfiguration/console

Reply:
@cshung this seems like something the GC team should investigate. I am unclear why this would be occurring but given the simplicity of the test I'd wager it is unlikely a test issue.

Reply:
@mangod9 It looks like the GCStandAloneServer set `COMPlus_gcServer=1`, so even when the code set `DOTNET_gcServer=0`, we are still unable to disable it.

Reply:
ok makes sense. @am11 what CI job was failing due to this issue? Is the failure only after this change or unrelated to this?

Reply:
@mangod9, it's a new test added by this PR, which started to fail in outerloop test when this PR was merged. Perhaps the root-cause was preexisting and the test has surfaced the problem.

> It looks like the GCStandAloneServer set `COMPlus_gcServer=1`, so even when the code set `DOTNET_gcServer=0`, we are still unable to disable it.

Based on the config parsing implementation:
https://github.com/dotnet/runtime/blob/4da6b9a8d55913c0ea560d63590d35dc942425be/src/coreclr/inc/clrconfignocache.h#L75-L86
first `DOTNET_gcServer` will be checked and if the value is `NULL`, it will fallback to legacy, `COMPlus_gcServer`. It is `LPCSTR`, so `=0` shouldn't be confused by NULL/0, right?